File: service/src/main/java/org/cadadr/conjugaison/controller/VerbConjugationController.java
package org.cadadr.conjugaison.controller;

import org.cadadr.conjugaison.domain.VerbConjugation;
import org.cadadr.conjugaison.service.VerbConjugationService;
import org.jetbrains.annotations.NotNull;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.graphql.data.method.annotation.Argument;
import org.springframework.graphql.data.method.annotation.QueryMapping;
import org.springframework.stereotype.Controller;

import java.util.List;

@Controller
public class VerbConjugationController {

    private final VerbConjugationService verbConjugationService;

    @Autowired
    public VerbConjugationController(@NotNull VerbConjugationService verbConjugationService) {
        this.verbConjugationService = verbConjugationService;
    }

    @QueryMapping
    public VerbConjugation conjugate(@Argument String infinitif) {
        return verbConjugationService.getConjugation(infinitif)
                .orElseThrow(() -> new VerbNotFoundException(infinitif));
    }

    @QueryMapping
    public @NotNull List<String> verbs() {
        return verbConjugationService.getVerbs();
    }

}

File: service/src/main/kotlin/org/cadadr/conjugaison/controller/VerbConjugationController.kt
package org.cadadr.conjugaison.controller

import org.cadadr.conjugaison.domain.VerbConjugation
import org.cadadr.conjugaison.service.VerbConjugationService
import org.springframework.graphql.data.method.annotation.Argument
import org.springframework.graphql.data.method.annotation.QueryMapping
import org.springframework.stereotype.Controller

@Controller
class VerbConjugationController(
    private val verbConjugationService: VerbConjugationService
) {

    @QueryMapping
    fun conjugate(@Argument verb: String): VerbConjugation =
        verbConjugationService.getConjugation(verb) ?: throw VerbNotFoundException(verb)

    @QueryMapping
    fun verbs(): List<String> = verbConjugationService.getVerbs()

}
